Detector/DetSensitive/src/DigitalCalorimeterSD.cpp

```diff
@@ -0,0 +1,400 @@
+#include "DetSensitive/DigitalCalorimeterSD.h"
+
+// FCCSW
+#include "DDSegmentation/BitField64.h"
+#include "DetCommon/DetUtils.h"
+#include "DDSegmentation/Segmentation.h"
+#include "DDSegmentation/CartesianGridXYZ.h"
+
+// DD4hep
+#include "DDG4/Geant4Mapping.h"
+#include "DDG4/Geant4VolumeManager.h"
+
+
+// CLHEP
+#include "CLHEP/Vector/ThreeVector.h"
+
+// Geant4
+#include "G4SDManager.hh"
+#include "G4ThreeVector.hh"
+#include "G4SystemOfUnits.hh"
+
+#include "iostream"
+#include "map"
+#include "unordered_map"
+
+namespace det {
+DigitalCalorimeterSD::DigitalCalorimeterSD(const std::string& aDetectorName,
+  const std::string& aReadoutName,
+  const dd4hep::Segmentation& aSeg)
+  : G4VSensitiveDetector(aDetectorName), m_seg(aSeg), m_calorimeterCollection(nullptr) {
+  // name of the collection of hits is determined byt the readout name (from XML)
+  collectionName.insert(aReadoutName);
+
+  m_headerPrinted = false;
+  m_countParticlesPerPixel = true;
+
+}
+
+DigitalCalorimeterSD::~DigitalCalorimeterSD(){
+}
+
+void DigitalCalorimeterSD::Initialize(G4HCofThisEvent* aHitsCollections)
+{
+  // create a collection of hits and add it to G4HCofThisEvent
+  // deleted in ~G4Event
+  m_calorimeterCollection = new G4THitsCollection
+    <dd4hep::sim::Geant4CalorimeterHit>(SensitiveDetectorName,collectionName[0]);
+  aHitsCollections->AddHitsCollection(G4SDManager::GetSDMpointer()->GetCollectionID(m_calorimeterCollection), m_calorimeterCollection);
+
+  //this collection is just a temp and not added to the HitsCollection just used to store raw hits before summing together
+  m_tempCollection = new G4THitsCollection
+    <dd4hep::sim::Geant4CalorimeterHit>(SensitiveDetectorName,"temp");
+
+  // m_pixelsOverThresholdFile.open("pixelsOverThreshold.txt", std::ios_base::app);.
+
+  if(!m_headerPrinted){
+    if(m_countParticlesPerPixel) {
+        std::cout << "\n\n\n###############################" << std::endl;
```

Comment:
If you need to print out information, you should rather use `MsgStream` e.g. as in your detector construction.